dbms/src/Flash/Coprocessor/DAGResponseWriter.cpp

```diff
@@ -120,17 +120,13 @@ void DAGResponseWriter::addExecuteSummaries(tipb::SelectResponse & response, boo
                 auto it = dag_context.getJoinExecuteInfoMap().find(join_executor_id);
                 if (it != dag_context.getJoinExecuteInfoMap().end())
                 {
-                    auto build_side_it = dag_context.getProfileStreamsMap().find(it->second.build_side_root_executor_id);
-                    if (build_side_it != dag_context.getProfileStreamsMap().end())
+                    UInt64 process_time_for_build = 0;
+                    for (const auto & join_build_stream : it->second.join_build_streams)
                     {
-                        UInt64 process_time_for_build = 0;
-                        for (const auto & join_build_stream : build_side_it->second)
-                        {
-                            if (auto * p_stream = dynamic_cast<IProfilingBlockInputStream *>(join_build_stream.get()))
-                                process_time_for_build = std::max(process_time_for_build, p_stream->getProfileInfo().execution_time);
-                        }
-                        current.time_processed_ns += process_time_for_build;
+                        if (auto * p_stream = dynamic_cast<IProfilingBlockInputStream *>(join_build_stream.get()); p_stream)
+                            process_time_for_build = std::max(process_time_for_build, p_stream->getProfileInfo().execution_time);
                     }
+                    current.time_processed_ns += process_time_for_build;
```

Comment:
Consider the case that the MPPTask has multiple joins, then it have to wait all build fininshed, does `join_build_streams` include all the join streams it had to wait?

Reply:
`dag_context.getExecutorIdToJoinIdMap().find(executor_id)` will save all join_executor_id in the child